sklearn/ensemble/gradient_boosting.py

```diff
@@ -965,8 +966,41 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # fit initial model - FIXME make sample_weight optional
             self.init_.fit(X, y, sample_weight)
 
-            # init predictions
-            y_pred = self.init_.predict(X)
+            if is_classifier(self.init_): 
+                n_classes = np.unique(y).shape[0]
+            else:
+                n_classes = 1
+
+            # If the initialization estimator has a predict_proba method,
+            # either use those, or collapse to a single vector if there
+            # are only two classes
+            if hasattr(self.init_, 'predict_proba'):
+                eps = np.finfo(X.dtype).eps
+                y_pred = self.init_.predict_proba(X) + eps
+                if n_classes == 2:
+                    y_pred = np.log(y_pred[:,1] / y_pred[:,0])
```

Comment:
Please fix the pep8 and add an inline comment to state explicitly that the binomial loss used for binary classification problems expects the predicted log odds rather than the predicted positive class probability.